### PR TITLE
[14.0][ADD] hr_timesheet_employee_analytic_tag: New addon

### PR DESCRIPTION
**Pull request by victoralmau**
New addon

Please @pedrobaeza and @LudLaf can you review it?

@Tecnativa TT39712

### CODE REVIEW

File: hr_timesheet_employee_analytic_tag/README.rst [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/timesheet-14-0/timesheet-14-0-hr_timesheet_employee_analytic_tag
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/117/14.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+This module allows define analytical labels in employees to be set when creating analytical items.
```
Comment (pedrobaeza) [inline hr_timesheet_employee_analytic_tag/README.rst]: Remember to put the `to`. It's `allow something` when the following word is a noun (example: `allow the entrance`), but when you use it with a verb, you need the `to` to "convert" such verb into something similar to a noun).

```suggestion
This module allows to define analytical labels in employees to be set when creating analytical items.
```

File: hr_timesheet_employee_analytic_tag/README.rst [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+.. contents::
+   :local:
+
+Configuration
+=============
+
+To configure this module, you need to:
+
+#. Go to *Settings > Users & Companies > Users*.
+#. Edit 'Mitchell Admin' user and set the following permissions:
```
Comment (pedrobaeza) [inline hr_timesheet_employee_analytic_tag/README.rst]: This has been commented other times: don't use specific nor demo data to put the steps. You have to redact in a generic way: `Put to your user the following permissions to see all the required elements: * Analytic...`

File: hr_timesheet_employee_analytic_tag/README.rst [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    * Analytic Accounting
+    * Analytic Accounting Tags
+    * Show Full Accounting Features
+
+Usage
+=====
+
+#. Go to *Employees > Employees* and edit 'Mitchell Admin' employee.
```
Comment (pedrobaeza) [inline hr_timesheet_employee_analytic_tag/README.rst]: The same about this.

File: hr_timesheet_employee_analytic_tag/README.rst [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+#. Set some 'Analytic Tags' in 'HR Settings' employee tab.
```
Comment (pedrobaeza) [inline hr_timesheet_employee_analytic_tag/README.rst]: This is part of configuration as well.

File: hr_timesheet_employee_analytic_tag/models/hr_employee.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class HrEmployee(models.Model):
+    _inherit = "hr.employee"
+
+    analytic_tag_ids = fields.Many2many(
```
Comment (pedrobaeza) [inline hr_timesheet_employee_analytic_tag/models/hr_employee.py]: Preventively avoid collapsing with other modules adding analytic tags to other things:

```suggestion
    timesheet_analytic_tag_ids = fields.Many2many(
```

File: hr_timesheet_employee_analytic_tag/tests/test_hr_timesheet_employee_analytic_tag.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        cls.account_analytic_line_model = cls.env["account.analytic.line"]
+
+    def _create_account_analytic_line(self, employee, tag):
+        vals = {
+            "name": "TEST",
+            "employee_id": employee.id,
+            "project_id": self.project.id,
+            "task_id": self.task.id,
+            "amount": 10,
+            "tag_ids": tag.ids if tag else False,
```
Comment (pedrobaeza) [inline hr_timesheet_employee_analytic_tag/tests/test_hr_timesheet_employee_analytic_tag.py]: Remove this line

File: hr_timesheet_employee_analytic_tag/views/hr_employee_view.xml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+            name="inherit_id"
+            ref="hr_timesheet.hr_employee_view_form_inherit_timesheet"
+        />
+        <field
+            name="groups_id"
+            eval="[(4, ref('analytic.group_analytic_accounting'))]"
+        />
+        <field name="arch" type="xml">
+            <group name="timesheet" position="inside">
+                <field name="analytic_tag_ids" widget="many2many_tags" />
```
Comment (pedrobaeza) [inline hr_timesheet_employee_analytic_tag/views/hr_employee_view.xml]: Missing analytic tags security group.

File: hr_timesheet_employee_analytic_tag/models/account_analytic_line.py [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    _inherit = "account.analytic.line"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        employee_model = self.env["hr.employee"]
+        for vals in vals_list:
+            if vals.get("employee_id"):
+                employee = employee_model.browse(vals.get("employee_id"))
+                if employee.analytic_tag_ids:
+                    vals["tag_ids"] = vals.get("tag_ids") or []
```
Comment (pedrobaeza) [inline hr_timesheet_employee_analytic_tag/models/account_analytic_line.py]:
```suggestion
                    vals["tag_ids"] = vals.get("tag_ids", [])
```

File: hr_timesheet_employee_analytic_tag/models/account_analytic_line.py [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        employee_model = self.env["hr.employee"]
+        for vals in vals_list:
+            if vals.get("employee_id"):
```
Comment (pedrobaeza) [inline hr_timesheet_employee_analytic_tag/models/account_analytic_line.py]: You should test as well if there's a project for saying that is a timesheet:
```suggestion
            if vals.get("employee_id") and vals.get("project_id"):
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
You should test with a `Form` the timesheets from a task.

**Comment by victoralmau** [comment]
Suggested changes done and test added with `Form()`

File: hr_timesheet_employee_analytic_tag/models/account_analytic_line.py [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+class AccountAnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        lines = super().create(vals_list)
+        for line in lines:
+            if line.project_id and line.employee_id:
+                employee = line.employee_id.sudo()
+                line.tag_ids += employee.timesheet_analytic_tag_ids
```
Comment (pedrobaeza) [inline hr_timesheet_employee_analytic_tag/models/account_analytic_line.py]: Why write after create? This is less efficient?

Comment (victoralmau) [reply hr_timesheet_employee_analytic_tag/models/account_analytic_line.py]: Sorry, it has been changed. In the test with Form() an incorrect view was being used and the employee was not defined until after creation.

File: hr_timesheet_employee_analytic_tag/readme/CONFIGURE.rst [hunk 11]
```diff
@@ -0,0 +1,10 @@
+To configure this module, you need to:
+
+#. Go to *Settings > Users & Companies > Users*.
+#. Put to your user the following permissions to see all the required elements:
```
Comment (LudLaf) [inline hr_timesheet_employee_analytic_tag/readme/CONFIGURE.rst]: #. Give your user these technical permissions to see all required elements:

File: hr_timesheet_employee_analytic_tag/readme/CONFIGURE.rst [hunk 12]
```diff
@@ -0,0 +1,10 @@
...
+    * Analytic Accounting
+    * Analytic Accounting Tags
+    * Show Full Accounting Features
+
+#. Go to *Employees > Employees* and edit your employee.
```
Comment (LudLaf) [inline hr_timesheet_employee_analytic_tag/readme/CONFIGURE.rst]: #. Go to the *Employees* app and edit your employee (should match your user).

File: hr_timesheet_employee_analytic_tag/readme/CONFIGURE.rst [hunk 13]
```diff
@@ -0,0 +1,10 @@
...
+#. Set some 'Analytic Tags' in 'HR Settings' employee tab.
```
Comment (LudLaf) [inline hr_timesheet_employee_analytic_tag/readme/CONFIGURE.rst]: #. Choose or create 'Analytic Tags' in 'Timesheets' under 'HR Settings' tab.

File: hr_timesheet_employee_analytic_tag/readme/USAGE.rst [hunk 14]
```diff
@@ -0,0 +1,3 @@
+#. Go to *Timesheets* and creates a new record.
+#. Go to *Invoicing > Configuration > Management > Analytic Items*.
+#. The previously created record will have the employee's analytical tags defined.
```
Comment (LudLaf) [inline hr_timesheet_employee_analytic_tag/readme/USAGE.rst]: #. The previously created record will have the defined employee's analytical tags.

**Comment by victoralmau** [comment]
Changes done.

**Review by pedrobaeza — APPROVED** [review]

**Review by sergio-teruel — APPROVED** [review]

**Pull request merged**